src/types/UiaTextRangeBase.cpp

```diff
@@ -372,22 +677,168 @@ try
+    {
+        pRetVal->vt = VT_I4;
+        pRetVal->iVal = base::ClampedNumeric<short>(attr.GetForeground().GetRGB());
+        return _getAttrVerificationFn(attributeId, *pRetVal);
+    }
+    case UIA_IsItalicAttributeId:
+    {
+        pRetVal->vt = VT_BOOL;
+        pRetVal->iVal = attr.IsItalic();
+        return _getAttrVerificationFn(attributeId, *pRetVal);
+    }
+    case UIA_StrikethroughColorAttributeId:
+    {
+        // Default Value: 0
+        // Source: https://docs.microsoft.com/en-us/windows/win32/winauto/uiauto-textattribute-ids
+        pRetVal->vt = VT_I4;
+        pRetVal->iVal = base::ClampedNumeric<short>(attr.IsCrossedOut() ? attr.GetForeground().GetRGB() : 0);
+        return _getAttrVerificationFn(attributeId, *pRetVal);
+    }
+    case UIA_StrikethroughStyleAttributeId:
+    {
+        pRetVal->vt = VT_I4;
+        pRetVal->iVal = static_cast<short>(attr.IsCrossedOut() ? TextDecorationLineStyle_Single : TextDecorationLineStyle_None);
+        return _getAttrVerificationFn(attributeId, *pRetVal);
+    }
+    case UIA_UnderlineColorAttributeId:
+    {
+        // Default Value: 0
+        // Source: https://docs.microsoft.com/en-us/windows/win32/winauto/uiauto-textattribute-ids
+        pRetVal->vt = VT_I4;
+        pRetVal->iVal = attr.IsUnderlined() ? base::ClampedNumeric<short>(attr.GetForeground().GetRGB()) : 0;
+        return _getAttrVerificationFn(attributeId, *pRetVal);
+    }
+    case UIA_UnderlineStyleAttributeId:
+    {
+        pRetVal->vt = VT_I4;
+        pRetVal->iVal = static_cast<short>(attr.IsUnderlined() ? TextDecorationLineStyle_Single : TextDecorationLineStyle_None);
+        return _getAttrVerificationFn(attributeId, *pRetVal);
+    }
+    default:
+        // This attribute is not supported.
+        pRetVal->vt = VT_UNKNOWN;
+        UiaGetReservedNotSupportedValue(&pRetVal->punkVal);
+        return nullptr;
+    }
+}
+
+IFACEMETHODIMP UiaTextRangeBase::GetAttributeValue(_In_ TEXTATTRIBUTEID attributeId,
                                                    _Out_ VARIANT* pRetVal) noexcept
 {
     RETURN_HR_IF(E_INVALIDARG, pRetVal == nullptr);
 
-    if (textAttributeId == UIA_IsReadOnlyAttributeId)
+    // AttributeIDs that require special handling
+    switch (attributeId)
+    {
+    case UIA_FontNameAttributeId:
+    {
+        pRetVal->vt = VT_BSTR;
+        pRetVal->bstrVal = SysAllocString(_pData->GetFontInfo().GetFaceName().data());
+        UiaTracing::TextRange::GetAttributeValue(*this, attributeId, *pRetVal);
+        return S_OK;
+    }
+    case UIA_FontSizeAttributeId:
+    {
+        pRetVal->vt = VT_R8;
+        // TODO CARLOS: how do I get the font size in points?
+        pRetVal->dblVal = 0; //_pData->GetFontInfo().GetUnscaledSize())
+        UiaTracing::TextRange::GetAttributeValue(*this, attributeId, *pRetVal);
+        return S_OK;
+    }
+    case UIA_IsReadOnlyAttributeId:
     {
         pRetVal->vt = VT_BOOL;
         pRetVal->boolVal = VARIANT_FALSE;

```

Comment:
well this rather depends: the entire buffer is technically "read-only"... it just happens to change when the connected application receives input.

A non-read-only buffer suggests that you could, e.g., select a region and delete it... or change its formatting or something.

Reply:
The Text Attribute ID docs:
> Identifies the IsReadOnly text attribute, which indicates whether the text is read-only (TRUE) or can be modified (FALSE).

@codeofdusk made an interesting suggestion as to the interpretation of that text: append "by the user" at the end. In console, a user can interact with the buffer via key strokes, and those key strokes may modify any section of the buffer. Surprisingly, something like "text on a webpage" is also _not_ read only! `IsReadOnly` seems to be limited to text that is disabled and cannot be interacted with in any way.